components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/AccountValidatorThread.java

```diff
@@ -179,22 +201,40 @@ private void disableAccounts(String tenantDomain, long suspensionDelay) throws I
             throw IdentityException.error("Error occurred while retrieving users for account disable", e);
         }
         if (receivers.size() > 0) {
-            try {
-                PrivilegedCarbonContext.startTenantFlow();
-                PrivilegedCarbonContext privilegedCarbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
-                privilegedCarbonContext.setTenantId(IdentityTenantUtil.getTenantId(tenantDomain));
-                privilegedCarbonContext.setTenantDomain(tenantDomain);
+            for (NotificationReceiver receiver : receivers) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Locking idle account: " + IdentityUtil.addDomainToName(receiver.getUsername(),
+                            receiver.getUserStoreDomain()) + "@" + tenantDomain);
+                }
+                RealmService realmService = NotificationTaskDataHolder.getInstance().getRealmService();
+                int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+
+                UserRealm userRealm;
+                try {
+                    userRealm = (UserRealm) realmService.getTenantUserRealm(tenantId);
+                } catch (UserStoreException e) {
+                    throw new IdentityException("Failed retrieve the user realm for tenant: " + tenantDomain, e);
+                }
 
-                UserIdentityManagementAdminService userIdentityManagementAdminService = new UserIdentityManagementAdminService();
-                // Iterate through all the receivers and disable them
-                for (NotificationReceiver receiver : receivers) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Disabling user " + receiver.getUsername());
-                    }
-                    userIdentityManagementAdminService.disableUserAccount(receiver.getUsername(), "email");
+                UserStoreManager userStoreManager;
+                try {
+                    userStoreManager = userRealm.getUserStoreManager();
+                } catch (org.wso2.carbon.user.core.UserStoreException e) {
```

Comment:
Shall we import the exception and use it here?

Reply:
org.wso2.carbon.user.api.UserStoreException is alredy imported.